compiler/qsc_parse/src/item.rs

```diff
@@ -149,6 +222,14 @@ fn parse_namespace(s: &mut ParserContext) -> Result<Namespace> {
     })
 }
 
+/// Parses the contents of a namespace block, what is in between the open and close braces in an
+/// explicit namespace, and any top level items in an implicit namespace.
+#[allow(clippy::vec_box)]
+fn parse_namespace_block_contents(s: &mut ParserContext) -> Result<Vec<Box<Item>>> {
```

Comment:
Can you just use `parse_many`? I assume the code was able to do this logic before your changes, as namespace blocks aren't new.